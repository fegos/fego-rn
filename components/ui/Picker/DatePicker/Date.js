/**
 * 日期选择器 DatePicker - 日期模式 组件
 * @author asyxu
 */
import React from 'react';
import PropTypes from 'prop-types';
import {

} from 'react-native';
import { UIComponent } from '../../../common';
import PickerView from '../PickerView';


export default class DatePicker extends UIComponent {
  static defaultProps = {
    initialValue: new Date(),
    minDate: new Date(),
    maxDate: new Date(new Date().getTime() + (1000 * 60 * 60 * 24 * 31 * 30)), // 默认多一年
    dateMode: 'year-month-day',
  }

  static propTypes = {
    // 初始时间
    initialValue: PropTypes.instanceOf(Date),
    // 最小日期
    minDate: PropTypes.instanceOf(Date),
    // 最大日期
    maxDate: PropTypes.instanceOf(Date),
    // 日期模式
    dateMode: PropTypes.oneOf(['year-only', 'month-only', 'day-only', 'year-month', 'year-month-day', 'month-day']),

  }
  constructor(props) {
    super(props);

    this._initDate(props);
    const { year, month, date } = this._handleInitialValue(props);
    this.state = {
      data: this._getInitialStateData(year, month, date),
      year,
      month,
      date,
    };
  }
  _handleInitialValue(props) {
    const { initialValue, minDate, maxDate } = props;
    const _initialValue = initialValue;
    // 解决日期输入格式
    let year = _initialValue.getFullYear();
    let month = _initialValue.getMonth();
    let date = _initialValue.getDate();

    // 有最小时最大时，需判断初始值是否合法
    if (minDate || maxDate) {
      let mYear;
      let mMonth;
      let mDate;
      if (minDate) {
        mYear = minDate.getFullYear();
        mMonth = minDate.getMonth();
        mDate = minDate.getDate();
      }

      // 保证在比较的时候只比较年月日，时间相同
      const iniT = new Date(year, month, date).getTime();
      const minT = minDate && (new Date(mYear, mMonth, mDate).getTime());
      const maxT = maxDate && (new Date(maxDate.getFullYear(), maxDate.getMonth(), maxDate.getDate()).getTime());
      /**
       * 初始值不在可选值范围内，直接设最小值为初始值
       * 若无最小值 minDate，则最小值取1970.01.01
       */
      if ((minDate && (iniT < minT)) || (maxDate && (iniT > maxT))) {
        if (minDate) {
          year = mYear;
          month = mMonth;
          date = mDate;
        } else {
          year = 1970;
          month = 0;
          date = 1;
        }
      }
    }
    return {
      year,
      month,
      date,
    };
  }
  _getInitialStateData(year, month, date) {
    return this._getDatePickerData(year, month, date);
  }
  _initDate(props) {
    const { minDate, maxDate } = props;
    let minYear;
    let minMonth;
    let _minDate;
    let maxYear;
    let maxMonth;
    let _maxDate;

    if (minDate) {
      minYear = minDate.getFullYear();
      minMonth = minDate.getMonth();
      _minDate = minDate.getDate();
    }
    if (maxDate) {
      maxYear = maxDate.getFullYear();
      maxMonth = maxDate.getMonth();
      _maxDate = maxDate.getDate();
    }

    this.minYear = minYear || 1970;
    this.minMonth = minMonth || 0;
    this.minDate = _minDate || 1;
    this.maxYear = maxYear || 2050;
    this.maxMonth = maxMonth || 11;
    this.maxDate = _maxDate;

    const yearArray = [];
    const monthArray = [];
    const dateArray = [];
    for (let i = this.minYear; i <= this.maxYear; i++) {
      yearArray.push({
        label: `${i}年`,
        value: `${i}`,
      });
    }
    for (let i = 0; i < 12; i++) {
      monthArray.push({
        label: `${i + 1}月`,
        value: `${i}`,
      });
    }
    for (let i = 1; i <= 31; i++) {
      dateArray.push({
        label: `${i}日`,
        value: `${i}`,
      });
    }
    this.yearArray = yearArray;
    this.monthArray = monthArray;
    this.dateArray = dateArray;
  }


  _onChange = (val, idx, label) => {
    const { valueArr, indexArr, labelArr } = this._updateData(val, idx, label);
    this.props.onChange(valueArr, indexArr, labelArr, 'date');
  }
  _updateData(val, idx, label) {
    let { year, month, date } = this.state;
    let yearIdx;
    let yearLabel;
    let monthIdx;
    let monthLabel;
    let dateIdx;
    let dateLabel;
    switch (this.props.dateMode) {
      case 'year-only':
        [year] = val;
        [yearIdx] = idx;
        [yearLabel] = label;
        break;
      case 'month-only':
        [month] = val;
        [monthIdx] = idx;
        [monthLabel] = label;
        break;
      case 'day-only':
        [date] = val;
        [dateIdx] = idx;
        [dateLabel] = label;
        break;
      case 'year-month':
        [year, month] = val;
        [yearIdx, monthIdx] = idx;
        [yearLabel, monthLabel] = label;
        break;
      case 'year-month-day':
        [year, month, date] = val;
        [yearIdx, monthIdx, dateIdx] = idx;
        [yearLabel, monthLabel, dateLabel] = label;
        break;
      case 'month-day':
        [month, date] = val;
        [monthIdx, dateIdx] = idx;
        [monthLabel, dateLabel] = label;
        break;
      default:
        break;
    }

    const newD = this._getDatePickerData(year, month, date);
    const v = [year, month, date];
    newD.forEach((data, i) => {
      const item = data.filter(d => d.value === v[i]);
      if (!item.length) {
        /**
        * 可选值范围的变化，导致picker数据发生了变化
        * 导致此前选中的那一项在新数据里没有了
         * 此时默认选中新数据的第一项
        */
        if (i === 0) {
          year = data[0].value;
          yearIdx = 0;
          yearLabel = data[0].label;
        } else if (i === 1) {
          month = data[0].value;
          monthIdx = 0;
          monthLabel = data[0].label;
        } else if (i === 2) {
          date = data[0].value;
          dateIdx = 0;
          dateLabel = data[0].label;
        }
      }
    });
    this.setState({
      data: newD,
      year: parseInt(year, 10),
      month: parseInt(month, 10),
      date: parseInt(date, 10),
    });

    return {
      valueArr: [year, month, date],
      indexArr: [yearIdx, monthIdx, dateIdx],
      labelArr: [yearLabel, monthLabel, dateLabel],
    };
  }
  _getDaysInMonth(year, month) {
    return new Date(year, month, 0).getDate();
  }
  _getDatePickerData(year, month) {
    const valueYear = parseInt(year, 10);
    const valueMonth = parseInt(month, 10);
    const maxDateThisMonth = this._getDaysInMonth(valueYear, valueMonth + 1);
    let monthStart; // slice 方法第一个参数，表示起始位置
    let monthEnd; // slice 方法第二个参数，表示结束位置的下一个位置
    let dateStart; // slice 方法第一个参数，表示起始位置
    let dateEnd; // slice 方法第二个参数，表示结束位置的下一个位置

    if (valueYear === this.minYear && valueYear === this.maxYear) {
      // 选中的年份 === 最小年份 === 最大年份，则月份的范围只能是 [minDate.minMonth, this.maxMonth]
      monthStart = this.minMonth;
      monthEnd = this.maxMonth + 1;

      if (valueMonth === this.minMonth && valueMonth === this.maxMonth) {
        // 选中的月份 === 最小月份 === 最大月份，则日期的范围只能是 [minDate.minDate, this.maxDate]
        dateStart = this.minDate - 1;
        dateEnd = this.maxDate || maxDateThisMonth;
      } else if (valueMonth === this.minMonth) {
        // 选中的月份 === 最小月份，则日期的范围只能是 [minDate.minDate, 月底]
        dateStart = this.minDate - 1;
        dateEnd = maxDateThisMonth;
      } else if (valueMonth === this.maxMonth) {
        // 选中的月份 === 最大月份，则日期的范围只能是 [1号, this.maxDate]
        dateStart = 0;
        dateEnd = this.maxDate || maxDateThisMonth;
      } else {
        // 其他情况 日期的范围只能是 [1号, 月底]
        dateStart = 0;
        dateEnd = maxDateThisMonth;
      }
    } else if (valueYear === this.minYear) {
      // 选中的年份 === 最小年份，则月份的范围只能是 [minDate.minMonth, 11] // month 范围是0-11
      monthStart = this.minMonth;
      monthEnd = 12;

      if (valueMonth === this.minMonth) {
        // 选中的月份 === 最小月份，则日期的范围只能是 [minDate.minDate, 月底]
        dateStart = this.minDate - 1;
        dateEnd = maxDateThisMonth;
      } else {
        // 其他情况 日期的范围只能是 [1号, 月底]
        dateStart = 0;
        dateEnd = maxDateThisMonth;
      }
    } else if (valueYear === this.maxYear) {
      // 选中的年份 === 最大年份，则月份的范围只能是 [1, maxDate.maxMonth]
      monthStart = 0;
      monthEnd = this.maxMonth + 1;

      if (valueMonth === this.maxMonth) {
        // 选中的月份 === 最大月份，则日期的范围只能是 [0, maxDate.maxDate]
        dateStart = 0;
        dateEnd = this.maxDate || maxDateThisMonth;
      } else {
        // 其他情况 日期的范围只能是 [1号, 月底]
        dateStart = 0;
        dateEnd = maxDateThisMonth;
      }
    } else {
      monthStart = 0;
      monthEnd = 12;
      dateStart = 0;
      dateEnd = maxDateThisMonth;
    }

    const monthArr = this.monthArray.slice(monthStart, monthEnd);
    const dateArr = this.dateArray.slice(dateStart, dateEnd);

    return [this.yearArray, monthArr, dateArr];
  }

  _dateModeData() {
    const { data } = this.state;
    let res = [];
    const { year, month, date } = this.state;
    switch (this.props.dateMode) {
      case 'year-only':
        res = data.slice(0, 1);
        this.renderValue = [year.toString()];
        break;
      case 'month-only':
        res = data.slice(1, 2);
        this.renderValue = [month.toString()];
        break;
      case 'day-only':
        res = data.slice(2, 3);
        this.renderValue = [date.toString()];
        break;
      case 'year-month':
        res = data.slice(0, 2);
        this.renderValue = [year.toString(), month.toString()];
        break;
      case 'year-month-day':
        res = data.slice(0, 3);
        this.renderValue = [year.toString(), month.toString(), date.toString()];
        break;
      case 'month-day':
        res = data.slice(1, 3);
        this.renderValue = [month.toString(), date.toString()];
        break;
      default:
        break;
    }
    this.renderData = res;
  }

  render() {
    // 在constructor里面处理了initialValue，这里取出initialValue防止传递给下面
    const {
      maskClosable, visible, initialValue, mode, minDate, maxDate, minuteStep, pickRef, ...rest
    } = this.props;
    this._dateModeData();
    return (
      <PickerView
        ref={pickRef}
        {...rest}
        data={this.renderData}
        visible={visible}
        value={this.renderValue}
        maskClosable={maskClosable}
        onChange={this._onChange}
      />
    );
  }
}
