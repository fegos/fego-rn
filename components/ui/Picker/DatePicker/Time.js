/**
 * 日期选择器 TimePicker - 时间模式 组件
 * @author asyxu
 */
import React from 'react';
import PropTypes from 'prop-types';
import {

} from 'react-native';
import { UIComponent } from '../../../common';
import PickerView from '../PickerView';

export default class TimePicker extends UIComponent {
  static defaultProps = {
    initialValue: new Date(),
    minuteStep: 1,
    minDate: new Date(),
    maxDate: new Date(new Date().getTime() + (1000 * 60 * 60 * 2)),
  }
  static propTypes = {
    // 初始时间
    initialValue: PropTypes.instanceOf(Date),
    // 最小步长
    minuteStep: PropTypes.number,
    // 最小日期
    minDate: PropTypes.instanceOf(Date),
    // 最大日期
    maxDate: PropTypes.instanceOf(Date),

  }
  constructor(props) {
    super(props);

    this._initDate(props);
    const { hour, minute } = this._handleInitialValue(props);

    this.state = {
      data: this._getInitialStateData(hour, minute),
      hour,
      minute,
    };
  }
  _handleInitialValue(props) {
    const {
      initialValue, minDate, maxDate, minuteStep,
    } = props;
    const _initialValue = initialValue;
    let hour = _initialValue.getHours();
    let minute = _initialValue.getMinutes();
    // 有最小时最大时，因此需判断初始值是否合法
    if (minDate || maxDate) {
      // 保证在比较的时候只比较时间，年月日相同
      const y = initialValue.getFullYear();
      const m = initialValue.getMonth();
      const d = initialValue.getDate();
      const minT = minDate && (new Date(y, m, d, minDate.getHours(), minDate.getMinutes()).getTime());
      const maxT = maxDate && (new Date(y, m, d, maxDate.getHours(), maxDate.getMinutes()).getTime());
      /**
       * 初始值不在可选值范围内，直接设最小值为初始值
       * 若无最小值 minDate，则最小值取0:00
       */
      if ((minDate && (initialValue.getTime() < minT)) || (maxDate && (initialValue.getTime() > maxT))) {
        if (minDate) {
          hour = minDate.getHours();
          minute = minDate.getMinutes();
        } else {
          hour = 0;
          minute = 0;
        }
      }
    }
    minute -= minute % minuteStep;
    return {
      hour,
      minute,
    };
  }
  _getInitialStateData(hour, minute) {
    return this._getTimePickerData(hour, minute);
  }
  _initDate(props) {
    const { minDate, maxDate, minuteStep } = props;
    let minHour;
    let minMinute;
    let maxHour;
    let maxMinute;
    // 最小日期存在，最小日期为最小日期与step内间隔
    if (minDate) {
      minHour = minDate.getHours();
      minMinute = minDate.getMinutes();
      const remainder = minMinute % minuteStep === 0 ? 0 : 1;
      minMinute += remainder;
    }
    // 最大日期存在，最大日期为最小日期与step内间隔
    if (maxDate) {
      maxHour = maxDate.getHours();
      maxMinute = maxDate.getMinutes();
      maxMinute -= maxMinute % minuteStep;
    }

    this.minHour = minHour || 0;
    this.minMinute = minMinute || 0;
    this.maxHour = maxHour || 23;
    this.maxMinute = maxMinute || 59;

    const hourArray = [];
    const minuteArray = [];
    for (let i = this.minHour; i <= this.maxHour; i++) {
      hourArray.push({
        label: `${i}时`,
        value: `${i}`,
      });
    }
    for (let i = 0; i <= 59; i++) {
      minuteArray.push({
        label: `${i}分`,
        value: `${i}`,
      });
    }

    this.hourArray = hourArray;
    this.minuteArray = minuteArray;
  }

  _onChange = (val, idx, label) => {
    const { valueArr, indexArr, labelArr } = this._updateData(val, idx, label);
    this.props.onChange(valueArr, indexArr, labelArr, 'time');
  }
  _updateData(val, idx, label) {
    let [hour, minute] = val;
    let [hourIdx, minuteIdx] = idx;
    let [hourLabel, minuteLabel] = label;
    const newD = this._getTimePickerData(hour, minute);

    newD.forEach((data, i) => {
      const item = data.filter(d => d.value === val[i]);
      if (!item.length) {
        /**
         * 级联数据维护，保证默认选中上一次的hour
         * 不存在的时候默认选择第一个
         */
        if (i === 0) {
          hour = data[0].value;
          hourIdx = 0;
          hourLabel = data[0].label;
        } else if (i === 1) {
          minute = data[0].value;
          minuteIdx = 0;
          minuteLabel = data[0].label;
        }
      }
    });
    this.setState({
      data: newD,
      hour,
      minute,
    });
    return {
      valueArr: [hour, minute],
      indexArr: [hourIdx, minuteIdx],
      labelArr: [hourLabel, minuteLabel],
    };
  }
  _getTimePickerData(hour) {
    const valueHour = parseInt(hour, 10);
    let minuteStart; // slice 方法第一个参数，表示起始位置
    let minuteEnd; // slice 方法第二个参数，表示结束位置的下一个位置

    // 分下标从0开始的，minuteEnd要表示下一个位置，所以有一个加1操作
    if (valueHour === this.minHour && valueHour === this.maxHour) {
      // 选中的时 === 最小时 === 最大时(即只有hour只有一个，那么minute决定minuteArr长度)，则分的范围只能是 [this.minMinute, this.maxMinute]
      minuteStart = this.minMinute;
      minuteEnd = this.maxMinute + 1;
    } else if (valueHour === this.minHour) {
      // 选中的时 === 最小时(即hour不止一个，那么minminute决定minuteArr长度)，则分的范围只能是 [this.minMinute, 60]
      minuteStart = this.minMinute;
      minuteEnd = 60;
    } else if (valueHour === this.maxHour) {
      // 选中的时 === 最大时(即hour不止一个，那么maxminute决定minuteArr长度)，则分的范围只能是 [0, this.maxMinute]
      minuteStart = 0;
      minuteEnd = this.maxMinute + 1;
    } else {
      // 中间时刻，默认所有minutes
      minuteStart = 0;
      minuteEnd = 60;
    }

    const minuteArr = this.minuteArray.slice(minuteStart, minuteEnd);

    return [this.hourArray, minuteArr];
  }

  render() {
    const {
      maskClosable, visible, minuteStep, initialValue, mode, minDate, maxDate, timeRef, ...rest
    } = this.props;
    const { hour, minute, data } = this.state;
    const _data1 = data[1].filter((d, i) => (i % minuteStep) === 0);
    return (
      <PickerView
        ref={timeRef}
        {...rest}
        data={[data[0], _data1]}
        visible={visible}
        value={[hour.toString(), minute.toString()]}
        maskClosable={maskClosable}
        onChange={this._onChange}
      />
    );
  }
}
