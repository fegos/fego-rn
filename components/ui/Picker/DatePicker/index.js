/**
 * 日期选择器 DatePicker 组件
 * @author asyxu
 */
import React from 'react';
import PropTypes from 'prop-types';
import {

} from 'react-native';
import { UIComponent } from '../../../common';
import DateModePicker from './Date';
import TimeModePicker from './Time';

// 非受控组件
export default class DatePicker extends UIComponent {
  static defaultProps = {
    visible: false,
    mode: 'date',
    minuteStep: 1,
    title: '请选择',
    okText: '确定',
    cancelText: '取消',
    showInModal: true,
    maskClosable: true,
    dateMode: 'year-month-day',
    format: () => { },
  }
  static propTypes = {
    // 是否可见，受控属性
    visible: PropTypes.bool,
    // 日期选择器模式，目前支持 'date', 'time'
    mode: PropTypes.oneOf(['date', 'time']),
    // 日期模式
    dateMode: PropTypes.oneOf(['year-only', 'month-only', 'day-only', 'year-month', 'year-month-day', 'month-day']),
    // picker 初始值
    initialValue: PropTypes.instanceOf(Date),
    // 可选的最小日期
    minDate: PropTypes.instanceOf(Date),
    // 可选的最大日期
    maxDate: PropTypes.instanceOf(Date),
    // 'time' 模式下的时间间隔
    minuteStep: PropTypes.number,
    // 模态框标题
    title: PropTypes.string,
    // 确定的文案
    okText: PropTypes.string,
    // 取消的文案
    cancelText: PropTypes.string,
    // 是否在弹层中显示，false 则直接展示在视图中，因此设为 false 时也可以通过 popup.show 来进行展示
    showInModal: PropTypes.bool,
    // 点击蒙层是否允许关闭
    maskClosable: PropTypes.bool,
    // 每列数据选择变化后的回调函数
    onChange: PropTypes.func,
  }


  _onChange = (val, idx, label, pickerType) => {
    const { onChange } = this.props;
    if (onChange instanceof Function) {
      onChange(val, idx, label, pickerType);
    }
  }

  render() {
    const { mode, dataPickRef } = this.props;
    const callbacks = {
      onChange: this._onChange,
    };

    switch (mode) {
      case 'date':
        return <DateModePicker pickRef={dataPickRef} {...this.props} {...callbacks} />;
      case 'time':
        return <TimeModePicker timeRef={dataPickRef} {...this.props} {...callbacks} />;
      default:
        return null;
    }
  }
}
