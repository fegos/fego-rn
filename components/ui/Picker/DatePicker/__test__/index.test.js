import React from 'react';
import Renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import DatePicker from '../index';

describe('DatePicker Tests', () => {
  it('renders correctly', () => {
    const tree = Renderer.create(<DatePicker
      mode="date"
      initialValue={new Date(2017, 11, 27, 16, 13, 8)}
      visible
    />).toJSON();

    const tree2 = Renderer.create(<DatePicker
      mode="time"
      initialValue={new Date(2017, 11, 27, 16, 13, 8)}
      visible
    />).toJSON();


    expect(tree).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
  });

  it('render different type', () => {
    const dateWrapper = shallow(<DatePicker
      mode="date"
      visible
    />);
    const timeWrapper = shallow(<DatePicker
      mode="time"
      visible
    />);
    expect(dateWrapper.name()).toBe('DatePicker');
    expect(timeWrapper.name()).toBe('TimePicker');
  });
});

