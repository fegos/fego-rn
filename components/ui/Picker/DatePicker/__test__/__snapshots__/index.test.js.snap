// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Tests renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "2019年",
            "textColor": undefined,
            "value": "2019",
          },
          Object {
            "label": "2020年",
            "textColor": undefined,
            "value": "2020",
          },
          Object {
            "label": "2021年",
            "textColor": undefined,
            "value": "2021",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "2月",
            "textColor": undefined,
            "value": "1",
          },
          Object {
            "label": "3月",
            "textColor": undefined,
            "value": "2",
          },
          Object {
            "label": "4月",
            "textColor": undefined,
            "value": "3",
          },
          Object {
            "label": "5月",
            "textColor": undefined,
            "value": "4",
          },
          Object {
            "label": "6月",
            "textColor": undefined,
            "value": "5",
          },
          Object {
            "label": "7月",
            "textColor": undefined,
            "value": "6",
          },
          Object {
            "label": "8月",
            "textColor": undefined,
            "value": "7",
          },
          Object {
            "label": "9月",
            "textColor": undefined,
            "value": "8",
          },
          Object {
            "label": "10月",
            "textColor": undefined,
            "value": "9",
          },
          Object {
            "label": "11月",
            "textColor": undefined,
            "value": "10",
          },
          Object {
            "label": "12月",
            "textColor": undefined,
            "value": "11",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "15日",
            "textColor": undefined,
            "value": "15",
          },
          Object {
            "label": "16日",
            "textColor": undefined,
            "value": "16",
          },
          Object {
            "label": "17日",
            "textColor": undefined,
            "value": "17",
          },
          Object {
            "label": "18日",
            "textColor": undefined,
            "value": "18",
          },
          Object {
            "label": "19日",
            "textColor": undefined,
            "value": "19",
          },
          Object {
            "label": "20日",
            "textColor": undefined,
            "value": "20",
          },
          Object {
            "label": "21日",
            "textColor": undefined,
            "value": "21",
          },
          Object {
            "label": "22日",
            "textColor": undefined,
            "value": "22",
          },
          Object {
            "label": "23日",
            "textColor": undefined,
            "value": "23",
          },
          Object {
            "label": "24日",
            "textColor": undefined,
            "value": "24",
          },
          Object {
            "label": "25日",
            "textColor": undefined,
            "value": "25",
          },
          Object {
            "label": "26日",
            "textColor": undefined,
            "value": "26",
          },
          Object {
            "label": "27日",
            "textColor": undefined,
            "value": "27",
          },
          Object {
            "label": "28日",
            "textColor": undefined,
            "value": "28",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`DatePicker Tests renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "16时",
            "textColor": undefined,
            "value": "16",
          },
          Object {
            "label": "17时",
            "textColor": undefined,
            "value": "17",
          },
          Object {
            "label": "18时",
            "textColor": undefined,
            "value": "18",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "13分",
            "textColor": undefined,
            "value": "13",
          },
          Object {
            "label": "14分",
            "textColor": undefined,
            "value": "14",
          },
          Object {
            "label": "15分",
            "textColor": undefined,
            "value": "15",
          },
          Object {
            "label": "16分",
            "textColor": undefined,
            "value": "16",
          },
          Object {
            "label": "17分",
            "textColor": undefined,
            "value": "17",
          },
          Object {
            "label": "18分",
            "textColor": undefined,
            "value": "18",
          },
          Object {
            "label": "19分",
            "textColor": undefined,
            "value": "19",
          },
          Object {
            "label": "20分",
            "textColor": undefined,
            "value": "20",
          },
          Object {
            "label": "21分",
            "textColor": undefined,
            "value": "21",
          },
          Object {
            "label": "22分",
            "textColor": undefined,
            "value": "22",
          },
          Object {
            "label": "23分",
            "textColor": undefined,
            "value": "23",
          },
          Object {
            "label": "24分",
            "textColor": undefined,
            "value": "24",
          },
          Object {
            "label": "25分",
            "textColor": undefined,
            "value": "25",
          },
          Object {
            "label": "26分",
            "textColor": undefined,
            "value": "26",
          },
          Object {
            "label": "27分",
            "textColor": undefined,
            "value": "27",
          },
          Object {
            "label": "28分",
            "textColor": undefined,
            "value": "28",
          },
          Object {
            "label": "29分",
            "textColor": undefined,
            "value": "29",
          },
          Object {
            "label": "30分",
            "textColor": undefined,
            "value": "30",
          },
          Object {
            "label": "31分",
            "textColor": undefined,
            "value": "31",
          },
          Object {
            "label": "32分",
            "textColor": undefined,
            "value": "32",
          },
          Object {
            "label": "33分",
            "textColor": undefined,
            "value": "33",
          },
          Object {
            "label": "34分",
            "textColor": undefined,
            "value": "34",
          },
          Object {
            "label": "35分",
            "textColor": undefined,
            "value": "35",
          },
          Object {
            "label": "36分",
            "textColor": undefined,
            "value": "36",
          },
          Object {
            "label": "37分",
            "textColor": undefined,
            "value": "37",
          },
          Object {
            "label": "38分",
            "textColor": undefined,
            "value": "38",
          },
          Object {
            "label": "39分",
            "textColor": undefined,
            "value": "39",
          },
          Object {
            "label": "40分",
            "textColor": undefined,
            "value": "40",
          },
          Object {
            "label": "41分",
            "textColor": undefined,
            "value": "41",
          },
          Object {
            "label": "42分",
            "textColor": undefined,
            "value": "42",
          },
          Object {
            "label": "43分",
            "textColor": undefined,
            "value": "43",
          },
          Object {
            "label": "44分",
            "textColor": undefined,
            "value": "44",
          },
          Object {
            "label": "45分",
            "textColor": undefined,
            "value": "45",
          },
          Object {
            "label": "46分",
            "textColor": undefined,
            "value": "46",
          },
          Object {
            "label": "47分",
            "textColor": undefined,
            "value": "47",
          },
          Object {
            "label": "48分",
            "textColor": undefined,
            "value": "48",
          },
          Object {
            "label": "49分",
            "textColor": undefined,
            "value": "49",
          },
          Object {
            "label": "50分",
            "textColor": undefined,
            "value": "50",
          },
          Object {
            "label": "51分",
            "textColor": undefined,
            "value": "51",
          },
          Object {
            "label": "52分",
            "textColor": undefined,
            "value": "52",
          },
          Object {
            "label": "53分",
            "textColor": undefined,
            "value": "53",
          },
          Object {
            "label": "54分",
            "textColor": undefined,
            "value": "54",
          },
          Object {
            "label": "55分",
            "textColor": undefined,
            "value": "55",
          },
          Object {
            "label": "56分",
            "textColor": undefined,
            "value": "56",
          },
          Object {
            "label": "57分",
            "textColor": undefined,
            "value": "57",
          },
          Object {
            "label": "58分",
            "textColor": undefined,
            "value": "58",
          },
          Object {
            "label": "59分",
            "textColor": undefined,
            "value": "59",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
